src/AbstractSQLRules2SQL.ts

```diff
@@ -939,6 +940,12 @@ const typeRules: Dictionary<MatchFn> = {
 		const date = DateValue(getAbstractSqlQuery(args, 0), indent);
 		return `DATE(${date})`;
 	},
+	DateTrunc: (args, indent) => {
+		checkArgs('DateTrunc', args, 2);
+		const field = Field(getAbstractSqlQuery(args, 0), indent);
+		const str = args[1];
+		return `date_trunc('${str}',${field})`;
+	},
```

Comment:
Can you add an
```ts
if (engine === Engines.postgres) {
```
check please and throw an error otherwise, since afaict only postgres supports the DATE_TRUNC function and not websql/mysql